skimage/transform/_warps.py

```diff
@@ -841,17 +841,19 @@ def warp(image, inverse_map, map_args={}, output_shape=None, order=1,
             matrix = np.linalg.inv(inverse_map.__self__.params)
 
         if matrix is not None:
-            matrix = matrix.astype(np.double)
+            matrix = matrix.astype(image.dtype)
+            ctype = 'float' if image.dtype.char == 'f' else 'double'
```

Comment:
I would really like it if you found a way to call these `float32` and `float64`. Mostly `float` is ambiguous, `double` not so much.

Reply:
I was happy to make it work without the explicit import here to cython: I did not knew that passing strings for specialization works :-D
Well, I will give it a try.

Reply:
Infortunatly, this doesn't work :-(  This is because `_warp_cy.__signatures__.keys()` only contains `"float"` and `"double"`.

I added a comment trying to alleviate the ambiguity:
```python
# Manage _warp_fast specialization according to input data
# type floating point precision (single or double)
ctype = 'float' if image.dtype.char == 'f' else 'double'
```

Reply:
It needs a sentence like:

> Contrary to Python, Cython `float` follows the meaning of `float` in C, and is equivalent to `float32`.